server/connectors/bitbucket_connector/api_client.py

```
@@ -332,26 +366,60 @@ def delete_branch(self, workspace, repo_slug, name):
-        """Get diff for a spec (commit hash, branch, or base..head range)."""
-        url = f"{BITBUCKET_API_BASE}/repositories/{workspace}/{repo_slug}/diff/{spec}"
+        """Get diff for a spec (commit hash, branch, or base..head range).
+
+        For a ``base..head`` range, each side is URL-encoded individually so
+        that refs containing reserved characters remain safe while the literal
+        ``..`` separator is preserved as a single path segment.
+        """
+        if ".." in spec:
+            parts = spec.split("..")
+            if len(parts) != 2:
+                raise ValueError("diff spec must contain exactly one '..' separator")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)